yt/frontends/flash/fields.py

```diff
@@ -171,21 +179,43 @@ def _nion(field, data):
         if ("flash", "abar") in self.field_list:
             self.add_output_field(("flash", "abar"),
                                   sampling_type="cell",
-                                  units="1")
-        else:
+                                  units="")
+        elif ("flash", "sumy") in self.field_list:
             def _abar(field, data):
                 return 1.0 / data["flash","sumy"]
             self.add_field(("flash","abar"),
                            sampling_type="cell",
                            function=_abar,
-                           units="1")
+                           units="")
+        elif "eos_singlespeciesa" in self.ds.parameters:
+            def _abar(field, data):
+                return data.ds.parameters["eos_singlespeciesa"]*data["index", "ones"]
+            self.add_field(("flash","abar"),
+                           sampling_type="cell",
+                           function=_abar,
+                           units="")
 
-        def _number_density(fields,data):
-            return (data["nele"]+data["nion"])
-        self.add_field(("gas","number_density"),
-                       sampling_type="cell",
-                       function=_number_density,
-                       units=unit_system["number_density"])
+        if ("flash", "abar") in self:
+
+            self.alias(("gas", "mean_molecular_weight"), ("flash", "abar"))
+
+            def _number_density(field, data):
+                Na_code = data.ds.quan(Na, '1/code_mass')
+                return data["flash", "dens"]*Na_code/data["flash", "abar"]
+
+            self.add_field(("gas", "number_density"),
+                           sampling_type="cell",
+                           function=_number_density,
+                           units=unit_system["number_density"])
+
+        if "ye" in self.field_list:
```

Comment:
I'm not that familiar with the FLASH format. Can you get both this condition and the one above (`if ("flash", "abar") in self`)? If so, you could have one number_density field overwriting another. Should this be an `elif`?